src/components/page/PageContent.js

```diff
@@ -114,54 +91,28 @@ class PageContent extends React.Component {
   }
 
   getPreviousPageJSX() {
-    const previousPage = this.props.page ? this.getAdjacentCategoryPage(this.props.page.previous, 'prev') : null
-
-    if (!previousPage) {
-      return null
-    }
-
-    const sameCategory =
-      previousPage &&
-      previousPage.categories &&
-      previousPage.categories.map(c => urlify(c.toLowerCase())).includes(this.props.category.toLowerCase())
-
-    let url
-    if (sameCategory) {
-      if (previousPage.url.endsWith('/')) {
-        // Chapter overview
-        url = buildCategoryUrl(this.props.category)
-      } else {
-        // Regular page
-        url = buildPageUrl(this.props.category, previousPage.title)
-      }
-    } else {
-      // Previous Chapter
-      url = buildPageUrl(firstCategoryNameOrUnknown(previousPage.categories), previousPage.title)
-    }
+    const { url, text } = this.getPreviousPageData()
 
     return (
       <Link to={url} className={'button is-info is-outlined is-uppercase'}>
         <div className={'icon'}>
           <i className={'fas fa-chevron-left'} />
         </div>
-        <span>{previousPage.title}</span>
+        <span>{text}</span>
       </Link>
     )
   }
 
   getAdjacentCategoryPage(nextPreviousContainer, info) {
-    if (nextPreviousContainer) {
-      const categoryIndex = this.props.page.categories
-        .map(cat => urlify(cat.toLowerCase()))
-        .indexOf(this.props.category.toLowerCase())
-      if (categoryIndex !== -1) {
-        return this.props.pages.find(
-          page => page.url.toLowerCase() === nextPreviousContainer[categoryIndex].toLowerCase(),
-        )
-      } else if (info === 'next') {
-        console.log('-1 next', this.props.page.categories, this.props.category)
-      }
-    }
+    return getAdjacentCategoryPage(this.props.page, this.props.category, this.props.pages, nextPreviousContainer, info)
+  }
+
+  getPreviousPageData() {
+    return getPreviousPageData(this.props.page, this.props.category, this.props.pages)
+  }
+
+  getNextPageData() {
+    return getNextPageData(this.props.page, this.props.category, this.props.pages)
```

Comment:
it's ok, but I don't see a real benefit in moving that in a function for itself.
could be a bit confusing because the function now is named like the one in helpers.

Reply:
I found that better in the beginning, but I understand that it makes it less clear. I will stick to regular argument passing.